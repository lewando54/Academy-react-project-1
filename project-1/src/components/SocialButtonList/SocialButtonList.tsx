import SocialButton from "../SocialButton/SocialButton";

function SocialButtonList(){
    const socials = [
        {id: 0, color: '#e90800', icon: 'google', href: 'https://google.pl/'},
        {id: 1, color: '#4768aa', icon: 'facebook', href: 'https://google.pl/'},
        {id: 2, color: '#0068c1', icon: 'linkedin', href: 'https://google.pl/'}
    ];

    const socialItems = socials.map(social => 
        <SocialButton key={social.id} color={social.color} href={social.href} icon={social.icon}/>
    )

    return (
        <div className="w-full flex justify-center gap-4">
            {socialItems}
        </div>
    )
}

export default SocialButtonList