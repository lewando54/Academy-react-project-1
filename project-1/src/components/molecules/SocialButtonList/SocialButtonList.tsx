import React from 'react'
import SocialButton, { iconProp } from '../../atoms/SocialButton/SocialButton'
import SocialButtonListStyle from './SocialButtonList.style'

export type TSocialsArray = Array<{ id: number, color: string, icon: iconProp, href: URL['href'], testId?: string }>
interface ISocialButtonList {
  socials: TSocialsArray
  testId: string
}

function SocialButtonList ({ socials, testId }: ISocialButtonList): React.ReactElement {
  const socialItems = socials.map(social =>
        <SocialButton key={social.id} color={social.color} href={social.href} icon={social.icon} testId={testId + social.id.toString()}/>
  )

  return (
        <SocialButtonListStyle data-testid={testId}>
            {socialItems}
        </SocialButtonListStyle>
  )
}

export default SocialButtonList
