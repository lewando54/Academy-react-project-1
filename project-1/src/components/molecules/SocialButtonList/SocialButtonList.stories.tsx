/* eslint-disable @typescript-eslint/consistent-type-assertions */
import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import SocialButtonList from './SocialButtonList'

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: SocialButtonList,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    socials: { control: 'array' }
  }

} as ComponentMeta<typeof SocialButtonList>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof SocialButtonList> = (args) => <SocialButtonList {...args} />

export const Primary = Template.bind({})

Primary.args = {
  socials: [
    { id: 0, color: '#e90800', icon: 'google', href: 'https://google.pl/' },
    { id: 1, color: '#4768aa', icon: 'facebook', href: 'https://google.pl/' },
    { id: 2, color: '#0068c1', icon: 'linkedin', href: 'https://google.pl/' }
  ]
}
// More on args: https://storybook.js.org/docs/react/writing-stories/args
