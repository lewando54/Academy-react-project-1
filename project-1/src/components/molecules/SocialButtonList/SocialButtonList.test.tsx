import React from 'react'
import '@testing-library/jest-dom'
import * as renderer from 'react-test-renderer'
import SocialButtonList, { TSocialsArray } from './SocialButtonList'

describe('SocialButtonList', () => {
  it('should render correctly', () => {
    const socials: TSocialsArray = [
      { id: 0, color: '#e90800', icon: 'google', href: 'https://google.pl/' },
      { id: 1, color: '#4768aa', icon: 'facebook', href: 'https://google.pl/' },
      { id: 2, color: '#0068c1', icon: 'linkedin', href: 'https://google.pl/' }
    ]

    const tree = renderer.create(<SocialButtonList socials={socials} testId='test'/>)
    expect(tree).toMatchSnapshot()
    // Check czy się tworzą elementy
  })
})
