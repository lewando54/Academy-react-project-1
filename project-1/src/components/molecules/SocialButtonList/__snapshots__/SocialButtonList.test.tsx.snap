// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButtonList should render correctly 1`] = `
<div
  className="sc-gswNZR bmbhTa"
  data-testid="test"
>
  <a
    className="sc-bcXHqe evFYBE"
    color="#e90800"
    data-testid="test0"
    href="https://google.pl/"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 488 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
      />
    </svg>
  </a>
  <a
    className="sc-bcXHqe dsbYWA"
    color="#4768aa"
    data-testid="test1"
    href="https://google.pl/"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
      />
    </svg>
  </a>
  <a
    className="sc-bcXHqe CiKOA"
    color="#0068c1"
    data-testid="test2"
    href="https://google.pl/"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
      />
    </svg>
  </a>
</div>
`;
